lib/echo/echo_client.rb

```diff
@@ -94,6 +94,20 @@ def create_order(granule_query, option_id, option_name, option_model, user_id, t
       #return {order_id: 1234, count: 2000}
       catalog_response = cmr_client.get_granules(granule_query, token)
       granules = catalog_response.body['feed']['entry']
+      order_info = get_order_information(granules.map {|g| g['id']}, token).body
+
+      # drop granules from the order if the selected order option is not one of the supported options by this granule
+      dropped_granules = []
+      Array.wrap(order_info).each do |info|
```

Comment:
Suggestions for cleaning up and making it faster (You don't necessarily need to do it this way):
    
    # Given order info JSON references, returns a Hash of granule ids to their options
    def order_info_to_granule_options(order_info)
      order_info.index_by { |info| info['catalog_item_ref']['id'] }
        .transform_values { |info| Array.wrap(info['option_definition_refs']) }
        .transform_values { |ref| ref['name'] }
    end

    def get_option_definition(id, token)
      ...
      granules_by_id = granules.index_by {|g| g['id']}
      granule_options = order_info_to_granule_options(order_info)
      included_granule_ids = granule_options.select {|k, v| v.include?(option_name)}
      excluded_granule_ids = granule_options.reject {|k, v| v.include?(option_name)}
      granules = granules_by_id.values_at(*included_granule_ids)
      dropped_granules = granules_by_id.values_at(*excluded_granule_ids)